src/frontend/src/optimizer/plan_node/stream_row_id_gen.rs

```diff
@@ -26,23 +26,41 @@ pub struct StreamRowIdGen {
     pub base: PlanBase<Stream>,
     input: PlanRef,
     row_id_index: usize,
+    custom_distribution: bool,
 }
 
 impl StreamRowIdGen {
     pub fn new(input: PlanRef, row_id_index: usize) -> Self {
-        let distribution = if input.append_only() {
+        if input.append_only() {
```

Comment:
I guess we can not check in the new function, instead, passing the flag is needed. DmlExecutor is non-append only.